pandas/core/groupby.py

```diff
@@ -1925,6 +1925,9 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
 
             # a passed Categorical
             elif isinstance(self.grouper, Categorical):
+                # fix bug #GH8868 sort=False being ignored in categorical groupby
+                if not self.sort:
+                    self.grouper = self.grouper.reorder_categories(self.grouper.unique())
```

Comment:
you can just do

```
self.grouper = self.grouper.copy()
self.grouper.ordered = self.sort
```

We discussed in another issue creating a ``set_ordered()`` method to do exactly this.

@jorisvandenbossche 
cc @JanSchultz
@shoyer